### DISCUSSION TITLE
Event<LogEventData> and ExceptionDetails

### DISCUSSION THREAD

**Discussion by cwgbittencourt** [datalust/seq-tickets#1819] [Help] [answered]
I'm not retrieving the values ​​of the ExceptionDetails property of the Event<LogEventData> object. I get as return: System.Collections.Generic.Dictionary\`2\[System.String,System.Object]
Could you point out in the documentation how to handle this situation? I've tried to do following other implementations but the result is the same. SEQ version 2023.1.8899 .Thanks for any help I may receive.

**Comment by liammclennan** [comment]
Hi @cwgbittencourt 

So that I can be more helpful please provide:

* how you are sending log events to Seq, including sample code
* which client you are using (serilog, nlog, etc)
* what you expect to see in Seq
* what you are seeing in Seq

**Comment by cwgbittencourt** [comment]
Liam,
thank you for trying to help me

*how you are sending log events to Seq, including sample code*

My problem is not recording in SEQ

![image](https://user-images.githubusercontent.com/55493968/232812817-36b4c363-08aa-4241-a0e0-2c7c0221c610.png)

*which client you are using (serilog, nlog, etc)*

Serilog

*what you expect to see in Seq*

{  "Ambiente": "DEV", "Origem": "Test Whatsapp",  "Projeto": "SEQ",
"Messagem": "Someone is testing the alert notification."   }

*what you are seeing in Seq*
![image](https://user-images.githubusercontent.com/55493968/232813055-5b6adb8f-9101-4f6c-b97e-a4a0b9b11249.png)

When I use “Send to app”, the information is sent as expected.
![image](https://user-images.githubusercontent.com/55493968/232813122-3938292e-7401-4b50-b09a-e87856474ae6.png)

Information received on Whatsapp.  With the exception of
ExceptionDetail(How to retrieve this value correctly? Could you provide
example?)
![image](https://user-images.githubusercontent.com/55493968/232813187-50b049cf-736c-4695-bd13-2cce5c764698.png)


The time setting below is for testing purposes only and will not be the
actual scenario. Variable values are not being returned
![image](https://user-images.githubusercontent.com/55493968/232813251-d1863831-fbe1-4768-8a61-c586b0a7ac68.png)


Information received on Whatsapp
![image](https://user-images.githubusercontent.com/55493968/232813344-d7cbfc21-0732-4cc0-885c-ca64bca2e193.png)

Because it works correctly in Send to app and not in Alert?



Em seg., 17 de abr. de 2023 às 18:56, Liam McLennan <
***@***.***> escreveu:

**Reply by nblumhardt** [reply]
Hi @cwgbittencourt,

What you're seeing is the usual output of calling `ToString()` on a dictionary. When the data reaches your app, it's going to be structured rather than converted into a string, so you'll have to either serialize it (e.g. to JSON) or pull values out with e.g. `((IReadOnlyDictionary<string, object>)exceptionDetail)["HRESULT"]` etc.

If you're able to post the code showing how you're retrieving the data and formatting it for Whatsapp, we should be able to offer some more specific advice.

(We've drawn the conclusion that you're building the Whatsapp plug-in; if that's not the case and it's freely available somewhere, please let us know.)

Best regards,
Nick

**Comment by cwgbittencourt** [comment]
Hi Nick,
Your information already helped to understand what is being returned.

I know it's not ideal but with this change I can already see what is returned.
`return JsonSerializer.Serialize(evt.Data.RenderedMessage) + "<br><br>" + string.Join("<br>", JsonSerializer.Serialize(evt.Data.Properties)) + "<br><br>" + JsonSerializer.Serialize(evt.Data.Exception);`

But I don't understand the following. When I press the "Send to App" button, the values ​​sent are those contained in the SEQ Events but when it is triggered directly in the Alert conditions, the values ​​are completely different. It looks like the Event properties are ignored.

`"Alert condition triggered by \u0022teseu/Alerta Whatsapp\u0022"<br>
<br>
{
"NamespacedAlertTitle":"teseu/Alerta Whatsapp",
"Alert":{	"Id":"alert-51",
		"Title":"Alerta Whatsapp",
		"Url":"http://localhost:5341/#/alerts/alert-51",
		"OwnerUsername":"teseu"
		,"SignalExpressionDescription":null,
		"Query":"select count(*) as count from stream where @Level = \u0027Warning\u0027 and Ambiente = \u0027DEV\u0027 group by time(1m) having count \u003E 0 limit 100 for background","HavingClause":"count \u003E 0",
		"TimeGrouping":"1 minute"
	},
"Source":{	"RangeStart":"2023-04-19T18:41:26.3478906Z",
		"RangeEnd":"2023-04-19T18:42:26.3478906Z",
		"ResultsUrl":"http://localhost:5341/#/events?q=select%20count%28%2A%29%20as%20count%20from%20stream%20where%20%40Level%20%3D%20%27Warning%27%20and%20Ambiente%20%3D%20%27DEV%27%20group%20by%20time%281m%29%20having%20count%20%3E%200%20limit%20100%20for%20background\u0026from=2023-04-19T18:41:26.3478906Z\u0026to=2023-04-19T18:42:26.3478906Z",
		"Results":[["time","count"],["2023-04-19T18:41:26.3478906Z",1]],
		"ContributingEvents":null},
"SuppressedUntil":"2023-04-19T18:43:56.3478906Z",
"Failures":null}
<br><br>null
`

And yes, we are creating a plug-in for sending messages to a Whatsapp Group, it is necessary to have contracted a specific third-party service. Today it is only in our internal environment but the company has already authorized it to be made available to the community. As soon as I adjust some things here I'll be letting you know.

**Reply by liammclennan** [reply, marked as answer]
Hi @cwgbittencourt 

Alerts are caused by events, but don't directly map to events. This is why sending an event to an app is not the same as an alert triggering an app. 

If you choose 'customize this notification' when configuring your alert then you can set 'Include contributing events' to include a sample of events with the alert. What is included with the alert is described in [the documentation](https://docs.datalust.co/docs/alert-properties).
